src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -194,6 +207,16 @@ public boolean isClosingWithinTimeLimit(long hours) {
                && difference.compareTo(Duration.ofHours(hours)) < 0;
     }
 
+    public boolean isOpeningWithinTimeLimit(long hours) {
+        Instant now = Instant.now();
+        Duration difference = Duration.between(now, startTime);
+
+        return now.isBefore(startTime)
+                && difference.compareTo(Duration.ofHours(hours - 1)) >= 0
```

Comment:
can we use `difference.compareTo(Duration.ofHours(hours)) > 0` to make it consistent the other comparison?

Reply:
@daongochieu2810 Hmm I believe that would make the function always return false? These three lines together return whether the current time is 23-24 hours before the session start time. I followed the `isClosingWithinTimeLimit` method that is right above but yes it may need changes

<details>
  <summary>Here's the context of where <code>isOpeningWithinTimeLimit</code> is used just in case it's confusing </summary>

   
```java
public List<FeedbackSessionAttributes> getFeedbackSessionsOpeningWithinTimeLimit() {
        List<FeedbackSessionAttributes> requiredSessions = new ArrayList<>();

        // this line returns emails that open anytime after 2 days before now that haven't had their opening soon email sent
        List<FeedbackSessionAttributes> sessions = fsDb.getFeedbackSessionsPossiblyNeedingOpeningSoonEmail();
        log.info(String.format("Number of sessions under consideration: %d", sessions.size()));

        for (FeedbackSessionAttributes session : sessions) {
            if (session.isOpeningWithinTimeLimit(NUMBER_OF_HOURS_BEFORE_OPENING_SOON_ALERT)
                    && !coursesLogic.getCourse(session.getCourseId()).isCourseDeleted()) {
                requiredSessions.add(session);
            }
        }
```

</details>

Reply:
@pPris Im not sure how `difference.compareTo(Duration.ofHours(hours)) > 0` is different than `difference.compareTo(Duration.ofHours(hours - 1)) >= 0`, can you elaborate?

Reply:
@ My understanding is that (assuming the value of the `hours` argument = 24)
the first one compares if difference greater than a duration of 24 hours, 
the second one compares if a difference is greater than or equal to 23 hours 

My understanding is that here, we wouldn't want to return true if the difference is greater than 24 hours. (we only want to send opening soon emails once the start time is less than 24 hours from now, which is what i believe this function should be doing?)

Reply:
> the first one compares if difference greater than a duration of 24 hours,
> the second one compares if a difference is greater than or equal to 23 hours

the 2nd one seems redundant to me then?

Reply:
yes ur right, lets leave this as it is then